app.js

```diff
@@ -23,9 +41,47 @@ app.use(cookieParser());
 app.use(bodyParser.json());
 app.use(express.static(path.join(__dirname, "public")));
 
+app.use(isAuth);
 app.use("/", indexRouter);
 app.use("/users", usersRouter);
 
+io.sockets.on("connection", function (socket, client) {
+  console.log("client connected!")
+  socket.on('room', function(room){
+    socket.join(room);
+    console.log(`Someone joined the room: ${room}`)
+  })
+  socket.on("newMessage", async (data) => {
+    try {
+        console.log(data);
+        Topic.findById(data.topicId, async function(err,Topic){
+          if(err){
+            throw new Error (err);
+          }
+          let user = await User.findById(data.userId, function(err,user){
```

Comment:
In is-auth.js you are storing some user info on req object so it would be better if you attach user info there and then access that by using req.user.username

Reply:
Oh nice!

Reply:
But can we use express middleware on sockets request? Like I have to make a socket.io middleware, let me know if there's a different approach. Thanks mate!

Reply:
@DevanshCodes Socket is onlying emitting events and responding to it and talk to client-side.
The request is made by the express only so if you are defining any routes for socket they yes middleware should work

Reply:
Resolved, will add differently middlewares